crates/miden-objects/src/block/block_inputs.rs

```diff
@@ -94,4 +94,39 @@ impl BlockInputs {
             self.unauthenticated_note_proofs,
         )
     }
+
+    // TESTING
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns a mutable reference to the [`ChainMmr`].
+    ///
+    /// Allows mutating the inner chain MMR for testing purposes.
+    #[cfg(any(feature = "testing", test))]
+    pub fn chain_mmr_mut(&mut self) -> &mut ChainMmr {
+        &mut self.chain_mmr
+    }
```

Comment:
Added these methods here because they can't be defined in the `testing` module without the fields being `pub(crate)`. Adding these here seemed cleaner, but we can go either way.